alphagov/paas-bootstrap
# [#156996834] Configure rsyslog to send logs to logit

## What

We want to send the logs to logit.io, so we need to configure the new target.

We also want to temporary keep sending the logs to the old stack. For that, we append a chunck of raw configuration to rsyslog. To do that, we need the changes included in the PR https://github.com/alphagov/paas-syslog-release/pull/3

**Regarding Mutual TLS**

Logit supports [mutual TLS](https://reliability-engineering.cloudapps.digital/manuals/logit-io.html#request-tls-mutual-authentication). We **did implement** this  the syslog-release PR https://github.com/alphagov/paas-syslog-release/pull/3 and in this PR

But later we have decided to not use mutual TLS, but IP whitelisting instead. The reason is:

> Mutual TLS carries some risks/complications around what happens when the client certs expire, and what the process for rotating the certs is. These certs have to be configured via a support ticket, and we're not clear what the response time is there, or whether we could have 2 certs live at once etc.
> 
> It's also not clear that there's really a need for it. With an open logstash endpoint, the worst that can happen is that someone discovers the endpoint and starts sending junk to it. I think the likelyhood of this is very low as there's not much to be gained by doing it.
> 
> For the staging and prod stacks, we can enable IP whitelisting for them, but we can pick that up in a follow-up story. For dev I think it's fine to leave it open.

Because that, we disabled mutual TLS for the time being, but we will merge the PR in syslog release anyway https://github.com/alphagov/paas-syslog-release/pull/3

For testing, we will keep two stacks in logit.io: one with mutual TLS and other one without.

## How to review

Test with mutual TLS (to test PR https://github.com/alphagov/paas-syslog-release/pull/3):

 1. Checkout the branch multiple-rsyslog-endpoints-156996834-2-mutual-tls
 2. Upload the logit secrets for mutual TLS:  
     - checkout the new secrets in the password store `paas git checkout  logit-secrets-156996834-mutual-tls`
     - Run `make dev upload-logit-secrets`
 3. Push this pipeline and deploy it. It is enough if you deploy concourse only to test this case
 4. You shall see the logs in both logit and the environment logsearch
 5. Once done, you can merge both the PRs:
     - https://github.com/alphagov/paas-syslog-release/pull/3
     - https://github.com/alphagov/paas-syslog-release/pull/4

Test without mutual TLS

 1. Once your merged the other PRs  https://github.com/alphagov/paas-syslog-release/pull/3 and https://github.com/alphagov/paas-syslog-release/pull/4, get the new release info in https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/syslog-release
 2. Update the corresponding in this PR (or ask @keymon to update it)
 3. Upload the logit secrets for NON mutual TLS
      - checkout the new secrets in the password store `paas git checkout  logit-secrets-156996834-mutual-tls`
     - Run `make dev upload-logit-secrets`
 4. Redeploy this branch: 
      - `create-bosh-concourse`
      - `create-cloudfoundry`
 5. Check that the logs work for both logit and the old logsearch

Once tested:
 - Merge 
   - https://github.com/alphagov/paas-credentials/pull/117
   - https://github.com/alphagov/paas-cf/pull/1360
   - https://github.com/alphagov/paas-logsearch-for-cloudfoundry/pull/7
  

## Before merging ensure that

 - [x] The logit for staging and prod is created
 - [x] The logstash filter are configured for staging and prod
 - [x]  we uploaded the logit secrets for prod and staging.
 - [x] alphagov/paas-syslog-release#4 has been merged, and the WIP commit updated.

## Who can review

Anyone but @keymon or @bandesz
